contracts/axelar-gas-service/src/contract.rs

```diff
@@ -29,31 +27,27 @@ impl AxelarGasService {
             .instance()
             .set(&DataKey::GasCollector, &gas_collector);
     }
-}
 
-#[contractimpl]
-impl AxelarGasServiceInterface for AxelarGasService {
     fn pay_gas_for_contract_call(
         env: Env,
         sender: Address,
         destination_chain: String,
         destination_address: String,
         payload: Bytes,
         refund_address: Address,
-        token_details: TokenDetails,
+        token: Token,
     ) {
         sender.require_auth();
 
-        let TokenDetails { token_addr, amount } = token_details.clone();
-
-        if amount == 0 {
+        if token.amount == 0 {
```

Comment:
```suggestion
        if token.amount <= 0 {
```
since it's a signed int